### PR TITLE
Dimaio/minimizer perf

### PR DESCRIPTION
**Pull request by fdimaio**
Fix the issue where DOF masking was leading to perf issues.

### CODE REVIEW

File: tmol/optimization/modules.py [hunk 1]
```diff
@@ -82,23 +66,29 @@ def __init__(self, score_system, dofs, kinforest, system_size, dof_mask=None):
...
 
+        if dof_mask is None:
+            dof_mask = torch.full(
+                dofs.shape[:-1], True, device=dofs.device, dtype=torch.bool
+            )
+
+        self.system_size = system_size
+
+        if dof_mask is None:
+            dof_mask = torch.full(
```
Comment (aleaverfay) [inline tmol/optimization/modules.py:77]: This is unreachable / duplicated. I think it's safe to remove.

File: tmol/optimization/sfxn_modules.py [hunk 2]
```diff
@@ -14,20 +13,23 @@ def __init__(
...
+        self.masked_coords = torch.nn.Parameter(self.full_coords[self.coord_mask])
         self.count = 0
 
     def forward(self):
         self.count += 1
-        self.full_coords = DOFMaskingFunc.apply(
-            self.masked_coords, self.coord_mask, self.full_coords
-        )
+        self.full_coords = self.full_coords.detach()
+        self.full_coords[self.coord_mask] = self.masked_coords
```
Comment (aleaverfay) [inline tmol/optimization/sfxn_modules.py:32]: So was the performance issue related to this scatter but that `.detach()` above resolves it?

**Review by aleaverfay — APPROVED** [review]
LGTM. Minor points in the comments.

File: tmol/optimization/modules.py (hunk 1 above)
Comment (fdimaio) [reply tmol/optimization/modules.py:77]: will do

File: tmol/optimization/sfxn_modules.py (hunk 2 above)
Comment (fdimaio) [reply tmol/optimization/sfxn_modules.py:32]: there is a little overhead with the scatter (the 0.55s v 0.65s difference we currently see) but most of the overhead was from the memory allocation
